data/scenarios/Challenges/Ranching/gated-paddock.yaml

```diff
@@ -0,0 +1,175 @@
+      - |
+        First, paddock the sheep so they don't drown.
+        Construct a fence using the nearby gate as a starting point.
+        Make sure there are no gaps!
+      - |
+        Note that you can use the "drill" command (by way of the "post puller" tool)
+        to demolish a fence that has been "placed".
+    condition: |
+      r <- robotnamed "sheep";
+      as r {run "scenarios/Challenges/Ranching/gated-paddock/enclosure-checking.sw"};
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - treads
+      - scanner
+      - dictionary
+      - branch predictor
+      - strange loop
+      - clock
+      - ADT calculator
+      - comparator
+      - workbench
+      - grabber
+      - lambda
+      - logger
+      - hearing aid
+      - post puller
+    inventory:
+      - [0, fence]
+      - [0, gate]
+      - [10, hinge]
+  - name: sheep
+    description:
+      - meandering livestock
+    display:
+      invisible: false
+      char: '@'
+    system: true
+    dir: [0, 1]
+    program: |
+      run "scenarios/Challenges/Ranching/gated-paddock/meandering-sheep.sw";
+entities:
+  - name: fence
+    display:
+      char: '#'
+    description:
+    - Keeps sheep in. And some other things out.
+    properties: [known, portable,  unwalkable]
+  - name: post puller
+    display:
+      char: 'P'
+      attr: rock
+    capabilities: [drill]
+    description:
+    - Good for dismantling fences.
+    properties: [known]
+  - name: scrap wood
+    display:
+      char: '\'
+    description:
+    - Scrap wood. Good for nothing, really.
+    properties: [known]
+  - name: gate
+    display:
+      char: '/'
+      attr: rock
+    description:
+    - A gate permits the player to pass through, but sheep cannot.
+    properties: [known]
+  - name: hinge
+    display:
+      char: 'U'
+      attr: rock
+    description:
+    - Facilitates swinging action.
+    properties: [known]
+  - name: cabin
+    display:
+      char: Π
+      attr: rock
+    description:
+    - Home sweet home.
+    properties: [known,  unwalkable]
+  - name: fresh breadcrumb
+    display:
+      char: '.'
+    description:
+    - A marker that can be put down and found again.
+    properties: [portable]
+  - name: treaded breadcrumb
+    display:
+      char: 'x'
+    description:
+    - A marker that can be put down and found again (for a second time).
+    properties: [portable]
+recipes:
+  - in:
+    - [2, board]
+    out:
+    - [1, fence]
+  - in:
+    - [0, fresh breadcrumb]
+    out:
+    - [1, fresh breadcrumb]
+  - in:
+    - [1, fresh breadcrumb]
+    out:
+    - [1, treaded breadcrumb]
+  - in:
+    - [1, fence]
+    out:
+    - [1, scrap wood]
+    required:
+    - [1, post puller]
+  - in:
+    - [1, hinge]
+    - [1, fence]
+    out:
+    - [1, gate]
+known: [mountain, tree, water]
+seed: 0
+solution: |
+  run "scenarios/Challenges/Ranching/gated-paddock/fence-construction.sw"
+world:
```

Comment:
@kostmo maybe add seed, so that the sheep always do the same thing.

I hope it is unlikely, but it could save us some headaches in the future if the sheep randomly came up with some clever escape plan. 🐑

Reply:
It would also make the solution more stable.  Currently, there is always the unlikely possibility the sheep will all decide to run straight into the ocean before you build the fence. :laugh:

Reply:
> add seed

Actually I have `seed: 0` already on line 136.  But I wonder if there's global influence on the random number generation that causes the sheep to behave non-reproducibly?  Perhaps robot-local random number generators would help?

Reply:
@kostmo Oh, my mistake then. 👍 The only other source of nondeterminism I can think of would be the order of elements in an unordered container. That may or may not have an effect here.

You can check that by listing the sheep location at the end and then testing that they are always the same
using `testSolution'`.

Reply:
> But I wonder if there's global influence on the random number generation that causes the sheep to behave non-reproducibly?

If there is, it's a bug.